src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -3,85 +3,101 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
 using OrchardCore.Entities;
 using OrchardCore.Modules;
 using OrchardCore.Users.Models;
 using OrchardCore.Users.TimeZone.Models;
 
-namespace OrchardCore.Users.TimeZone.Services
+namespace OrchardCore.Users.TimeZone.Services;
+
+public class UserTimeZoneService
 {
-    public class UserTimeZoneService
+    private const string EmptyTimeZone = "empty";
+    private const string CacheKey = "UserTimeZone/";
+    private static readonly DistributedCacheEntryOptions _slidingExpiration = new() { SlidingExpiration = TimeSpan.FromHours(1) };
+
+    private readonly IClock _clock;
+    private readonly IDistributedCache _distributedCache;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public UserTimeZoneService(
+        IClock clock,
+        IDistributedCache distributedCache,
+        IHttpContextAccessor httpContextAccessor
+    )
     {
-        private const string CacheKey = "UserTimeZone/";
-        private readonly TimeSpan _slidingExpiration = TimeSpan.FromMinutes(1);
-
-        private readonly IClock _clock;
-        private readonly IDistributedCache _distributedCache;
-        private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly UserManager<IUser> _userManager;
-
-        public UserTimeZoneService(
-            IClock clock,
-            IDistributedCache distributedCache,
-            IHttpContextAccessor httpContextAccessor,
-            UserManager<IUser> userManager
-            )
+        _clock = clock;
+        _distributedCache = distributedCache;
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public async ValueTask<ITimeZone> GetUserTimeZoneAsync()
+    {
+        var currentTimeZoneId = await GetCurrentUserTimeZoneIdAsync();
```

Comment:
Here we then need to check if timeZoneId == "empty"

Reply:
No, it's either null or a valid timezoneid

Reply:
Ah okay I've just seen that you return null then so all good.